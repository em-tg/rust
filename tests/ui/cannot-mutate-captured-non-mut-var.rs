//@ check-pass
#![feature(unboxed_closures, tuple_trait)]

use std::io::Read;

fn to_fn_once<A:std::marker::Tuple,F:FnOnce<A>>(f: F) -> F { f }

fn main() {
    let x = 1;
    to_fn_once(move|| { x = 2; });
    //~^ WARNING: cannot assign to `x`, as it is not declared as mutable

    let s = std::io::stdin();
    to_fn_once(move|| { s.read_to_end(&mut Vec::new()); });
    //~^ WARNING: cannot borrow `s` as mutable, as it is not declared as mutable
}
