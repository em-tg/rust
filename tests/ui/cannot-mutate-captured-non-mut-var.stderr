warning[E0594]: cannot assign to `x`, as it is not declared as mutable
  --> $DIR/cannot-mutate-captured-non-mut-var.rs:10:25
   |
LL |     to_fn_once(move|| { x = 2; });
   |                         ^^^^^ cannot assign
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider changing this to be mutable
   |
LL |     let mut x = 1;
   |         +++

warning[E0596]: cannot borrow `s` as mutable, as it is not declared as mutable
  --> $DIR/cannot-mutate-captured-non-mut-var.rs:14:25
   |
LL |     to_fn_once(move|| { s.read_to_end(&mut Vec::new()); });
   |                         ^ cannot borrow as mutable
   |
help: consider changing this to be mutable
   |
LL |     let mut s = std::io::stdin();
   |         +++

warning: 2 warnings emitted

Some errors have detailed explanations: E0594, E0596.
For more information about an error, try `rustc --explain E0594`.
